src/background/services/openPayments.ts

```diff
@@ -881,6 +882,43 @@ export class OpenPaymentsService {
     return outgoingPayment;
   }
 
+  async outgoingPaymentWaitForCompletion(
+    outgoingPaymentId: OutgoingPayment['id'],
+    {
+      signal,
+      maxAttempts = 10,
+    }: Partial<{ signal: AbortSignal; maxAttempts: number }> = {},
+  ) {
+    let attempt = 0;
+    let outgoingPayment: undefined | OutgoingPayment;
+    while (++attempt <= maxAttempts) {
+      signal?.throwIfAborted();
```

Comment:
Supported well enough for our use 🎉 
https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/throwIfAborted#browser_compatibility

<img src="https://github.com/user-attachments/assets/1369b6b8-686f-4914-b1d2-7c944a2bb123" width="400" alt="">